From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christian=20K=C3=B6nig?= <christian.koenig@amd.com>
Date: Fri, 26 Nov 2021 14:12:42 +0100
Subject: dma-buf: add DMA_RESV_USAGE_KERNEL v3

Add an usage for kernel submissions. Waiting for those are mandatory for
dynamic DMA-bufs.

As a precaution this patch also changes all occurrences where fences are
added as part of memory management in TTM, VMWGFX and i915 to use the
new value because it now becomes possible for drivers to ignore fences
with the WRITE usage.

v2: use "must" in documentation, fix whitespaces
v3: separate out some driver changes and better document why some
    changes should still be part of this patch.

Signed-off-by: Christian König <christian.koenig@amd.com>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Link: https://patchwork.freedesktop.org/patch/msgid/20220407085946.744568-5-christian.koenig@amd.com

(Backported from commit b29895e18304feb7e8afc6388db7ece60327b23c linux)

Reviewed-by: Arvind Yadav <arvind.yadav@intel.com>
Signed-off-by: Varun Gupta <varun.gupta@intel.com>
---
 include/linux/dma-resv.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/dma-resv.h b/include/linux/dma-resv.h
index e7f51dad6586..5b30f0b77b87 100644
--- a/include/linux/dma-resv.h
+++ b/include/linux/dma-resv.h
@@ -48,6 +48,7 @@
 extern struct ww_class reservation_ww_class;
 
 enum dma_resv_usage {
+	DMA_RESV_USAGE_KERNEL,
 	DMA_RESV_USAGE_WRITE,
 	DMA_RESV_USAGE_READ,
 };
-- 
2.43.0

